tidy3d/plugins/adjoint/components/simulation.py

```diff
@@ -0,0 +1,341 @@
+"""Defines a jax-compatible simulation."""
+from __future__ import annotations
+
+from typing import Tuple, Union
+from collections import namedtuple
+
+import pydantic as pd
+import numpy as np
+
+from jax.tree_util import register_pytree_node_class
+
+from ....components.base import cached_property
+from ....components.monitor import FieldMonitor, PermittivityMonitor
+from ....components.monitor import ModeMonitor, DiffractionMonitor
+from ....components.simulation import Simulation
+from ....components.data.monitor_data import FieldData, PermittivityData
+from ....components.types import Ax
+from ....log import log
+from ....constants import HERTZ
+
+from ..log import AdjointError
+from .base import JaxObject
+from .structure import JaxStructure
+from .geometry import JaxBox
+
+# used to store information when converting between jax and tidy3d
+JaxInfo = namedtuple(
+    "JaxInfo",
+    "num_input_structures num_output_monitors num_grad_monitors num_grad_eps_monitors "
+    "fwidth_adjoint",
+)
+
+# bandwidth of adjoint source in units of freq0 if no sources and no `fwidth_adjoint` specified
+FWIDTH_FACTOR = 1.0 / 10
+
+
+@register_pytree_node_class
```

Comment:
Yea I don't think the validation is the limiting factor, but it would be good, in general, to avoid this.  I just put together a proposal for this [here](https://github.com/flexcompute/tidy3d/issues/675), but I think it might need to wait a bit.